lib/elixir/lib/kernel.ex

```diff
@@ -3069,6 +3077,38 @@ defmodule Kernel do
     end
   end
 
+  # Called as ensure_evaled(list, {0, []}). Note acc is reversed.
+  def ensure_evaled(list, acc) do
+    :lists.mapfoldl(fn
+      {:|, meta, [head, tail]}, acc ->
+        {head, acc} = ensure_evaled_element(head, acc)
+        {tail, acc} = ensure_evaled_tail(tail, acc)
+      {{:|, meta, [head, tail]}, acc}
+      elem, acc ->
+        ensure_evaled_element(elem, acc)
+    end, acc, list)
+  end
+
+  defp ensure_evaled_element(elem, acc)
+       when is_number(elem) or is_atom(elem) or is_binary(elem) do
```

Comment:
My limit is whatever can fit on the screen with two files side-by-side. I guess I need to change font 😆